Feature: improve docs generation

ThisIsMissEm opened the pull request:
<!-- When fixing a bug: -->

I'd noticed that the vercel hosted doc previews were broken, this is work that aims to fix that as well as some other issues in working with the documentation.

![image](https://user-images.githubusercontent.com/30827/154754072-839a819b-2448-470b-bc23-14c795701d44.png)


- [ ] I've added a unit test to test for potential regressions of this bug. n/a
- [ ] The changelog has been updated, if applicable. n/a
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

See the commit message for details on what changed.

ThisIsMissEm committed:
docs: Stop publishing to gh-pages, we've vercel

matthieubosquet reviewed (approved):
LGTM

ThisIsMissEm commented:
I wasn't able to merge this yet because something is causing the source links to not work in vercel (they work locally though)

matthieubosquet commented:
By the way @ThisIsMissEm , is it correct that the publish-website.yml workflow (triggering on any push to main) is removed without replacement?

matthieubosquet commented:
Tagging @kay-kim as this will affect her docs generation workflow.

ThisIsMissEm commented:
> By the way @ThisIsMissEm , is it correct that the publish-website.yml workflow (triggering on any push to main) is removed without replacement?

yes, it's not needed as it's automatically deployed to vercel anyway, there shouldn't be a need to increase the repo size with doc builds

matthieubosquet commented:
I'm trying to double check if there is any intended audience for https://inrupt.github.io/solid-client-js/ or if we should just delete the gh-pages branch and kill gh-pages on this repo.

@ajacksified what do you think?
It could be a relatively cheap fallback, but if there's no action left to update it (publish-website), we should probably just kill it altogether, not good to have outdated stuff lingering.

kay-kim commented on docs/api/conf.py:
```diff
@@ -67,9 +67,9 @@
+html_theme_path = ['./docs-assets/themes']
 
-html_copy_source = False
+html_copy_source = True
```
Q: Curious, but why are we setting this option to True?

ThisIsMissEm replied:
Debugging, was trying to figure out the links issue

matthieubosquet commented:
@kay-kim OK, thank you for the precision.

Then unless @ajacksified you have more information, I would suggest we nuke the github pages and related branch altogether.

ThisIsMissEm commented:
I'm intending on doing that soon, I was just trying to figure out why I had a regression in the docs build on Vercel (Kim had noticed it when I was discussing it with her)

However, since it looks like (at present) it's a bug only in places where we do not have a repo url I'll finish rebasing and merging this soon.

> On 23. Feb 2022, at 19:10, Matthieu Bosquet ***@***.***> wrote:
> 
> ﻿
> @kay-kim OK, thank you for the precision.
> 
> Then unless @ajacksified you have more information, I would suggest we nuke the github pages and related branch altogether.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you were mentioned.

-- 
This e-mail, and any attachments thereto, is intended only for use by the 
addressee(s) named herein and may contain legally privileged, confidential 
and/or proprietary information. If you are not the intended recipient of 
this e-mail (or the person responsible for delivering this document to the 
intended recipient), please do not disseminate, distribute, print or copy 
this e-mail, or any attachment thereto. If you have received this e-mail in 
error, please respond to the individual sending the message, and 
permanently delete the email.

ajacksified commented:
Yup, let's nuke gh-pages

ThisIsMissEm committed:
docs: Improve generation of documentation

This is a fairly significant change, but fixes a bunch of things with our vercel-built documentation pages as well as improving the developer experience when working on documentation.

Scripts changes:
- npm run build-api-docs => npm docs:build
- npm run build-api-docs-preview-site => npm docs:build
- Added docs:clean (removes all generated files, typically you won't need this)
- Added docs:install (simplifies installation)

Improvements:
- Updated docs README with new instructions
- Moved docs-assets out of the build directory so it can be cached between builds, allowing faster docs generation (run docs:clean to update it)
- Added makefile command for generating the dist/ output
- Added warning if you try to run the makefile without sphinx-build being available, instead of crashing
- Pinned myst-parser and Sphinx dependency versions, as these broke the docs build due to myst-parser not yet having a major version.

ThisIsMissEm commented:
Merging without CI passing for E2E tests due to known issue.
